base/logger_file.go

```diff
@@ -205,20 +219,39 @@ func (lfc *FileLoggerConfig) init(ctx context.Context, level LogLevel, name stri
 		lfc.CollationBufferSize = &bufferSize
 	}
 
-	ticker := time.NewTicker(time.Hour)
+	var rotationTicker *time.Ticker
+	var rotationTickerCh <-chan time.Time
+	if i := lfc.Rotation.RotationInterval.Value(); i > 0 && rotateableLogger != nil {
+		rotationTicker = time.NewTicker(i)
+		rotationTickerCh = rotationTicker.C
+	}
+
+	logDeletionTicker := time.NewTicker(rotatedLogDeletionInterval)
 	go func() {
 		defer func() {
 			if panicked := recover(); panicked != nil {
-				WarnfCtx(ctx, "Panic when deleting rotated log files: \n %s", panicked, debug.Stack())
+				WarnfCtx(ctx, "Panic when deleting rotated log files: %s\n%s", panicked, debug.Stack())
 			}
 		}()
+		if rotationTicker != nil {
+			defer rotationTicker.Stop()
+		}
+		defer logDeletionTicker.Stop()
 		for {
 			select {
-			case <-ticker.C:
+			case <-ctx.Done():
+				return
+			case <-logDeletionTicker.C:
 				err := runLogDeletion(ctx, logFilePath, level.String(), int(float64(*lfc.Rotation.RotatedLogsSizeLimit)*rotatedLogsLowWatermarkMultiplier), *lfc.Rotation.RotatedLogsSizeLimit)
 				if err != nil {
 					WarnfCtx(ctx, "%s", err)
 				}
+			case <-rotationTickerCh:
```

Comment:
nit: I think this reads just a bit easier to see `rotationTicker.C` since I had to look a back to see if this was a normal ticker.

Reply:
I ended up writing it this way because `rotationTicker` can be nil if you haven't enabled a rotation interval. The select panics trying to access `C` on a nil ticker.

You _can_ however, safely select over a nil channel, and this avoids having to write a separate conditional select if you have log rotation enabled.